superset/views/core.py

```diff
@@ -94,36 +95,54 @@ def apply(self, query, func):  # noqa
 
 
 class DashboardFilter(SupersetFilter):
+    """
+    List dashboards with the following criteria:
+        1. Those which the user owns
+        2. Those which the user has favorited
+        3. Those which have been published (if they have access to at least one slice)
 
-    """List dashboards for which users have access to at least one slice or are owners"""
+    Show all dashboards for users which have the 'all_datasource_access' permission
```

Comment:
Ahh interesting, I wasn't aware most folks were using superset this way – I'll have it just use the other rules instead. The concern here was that there should be a way for admins at least to see all the dashboards so that they may be able to purge ones that were left for dead unpublished without having to go into the DB and set the dashboard inactive there. I just think it's better to not assume that those with  the superset instance are necessarily DB-savvy

So if you think it's a fine idea, I will do a check for the current user having the admin role and then show them all the dashboards in that case only. The other option would be to encapsulate this in a "can_see_all_dashboards" permission in case some folks delete the admin role.